parameterisation/k_epsilon.F90

```diff
@@ -647,7 +686,23 @@ subroutine keps_eddyvisc(state, advdif)
 
   ! Initialise viscosity to background value
   if (have_visc) then
-     call set(viscosity, bg_visc)
+     ewrite(1,*) "Entering Initialise Viscosity to Background Viscosity"   
+     ! Checking if the viscosity field is on a discontinuous mesh and the bg_viscosity is on continuous mesh
+     ! We need to remap before setting the field as the set subroutine does not remap automatically.
+     if (continuity(viscosity)<0 .and. continuity(bg_visc)>=0) then     
+        ewrite(1,*) "Entering background viscosity remap conditional"
+        call allocate(visc_dg, viscosity%mesh, "RemappedBackgroundViscosityDG")
+        call remap_field(bg_visc,visc_dg,stat)
+        if (stat/=0) then
+           FLExit('There was some problem remapping the continuous backgroud viscosity to discontinuous mesh')
```

Comment:
Someone might disagree with me here, but I think this should be an FLAbort instead?